tests/e2e/file_volume_statefulsets.go

```diff
@@ -0,0 +1,687 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fss "k8s.io/kubernetes/test/e2e/framework/statefulset"
+)
+
+var _ = ginkgo.Describe("[csi-file-vanilla] File Volume statefulset", func() {
```

Comment:
we could use the same tests from block vanilla and use if conditions in two places, why new script?

Reply:
This was discussed with Madan or Kavya before, as the block volume sts code has unnecessary code for the file volume tests - we decided to keep a dedicated file for the file volume sts.
example - volume attachment to node code was not required for the file tests
storage class creation needed a nfs4FSType - which is not needed for block
also, block sts manifest/yamls comes with the ReadOnlyOnce access mode and the file needed an RWX, and this needed additional condition checks in code
+1, the existing block sts has many other tests which are not required for this task and at places the logic to order of scale-up and scale-down changes for file tests which are not required for block.

Having all these tests in a single place makes code unmanageable - so we decided to keep it separate.